lib/ruby_terraform/command_line/boolean_value_functionality.rb

```diff
@@ -0,0 +1,17 @@
+module RubyTerraform
+  module CommandLine
+    module BooleanValueFunctionality
+      def assign_option_value(value)
+        @value = boolean_val(value)
+      end
+
+      def boolean_val(variable)
+        return nil if variable.nil?
+        return variable if variable.is_a?(TrueClass) || variable.is_a?(FalseClass)
+        return true if variable.respond_to?(:upcase) && variable[0].upcase == 'T'
```

Comment:
> * The switch overrides feels a bit clunky, I wonder if there is a way to lift
>   that up a level into the command itself or some mixin for a command so that
>   the command line builder stuff can be cleaner. It's only needed because of
>   some syntactic sugar provided for the commands themselves.

>   I also feel using `ActiveSupport::Inflector` might help clean this up too,
>   where maybe we are declarative about those options that can be repeated
>   and then we allow for a pluralized option name as an option in the hash passed
>   to the command.

I've had so many versions of trying to think of a good way to do this :-) The main driver is the `vars var_files targets` mapping which inflector would help with but there's also the two commands that want `directory -> -config` and `plan -> -out` which is why I ended up with thinking explicit declaration seemed like the only option that would cover all bases.

A mixin for the common array ones seems like a good possible option but I'm not sure about the two explicit mappings.. 

> * I'm trying to keep to 80 character line limit - would you mind formatting to
>   maintain that.

I have to ask - what development environment are you working in where 80 characters is a limit?  I'm only running 2160 x 1440 (haven't quite managed to pony up the pennies for 4k as yet) and I would imagine most people are at least on 1920 x 1080 these days.  80 characters takes me back to the Apple II days on a green screen haha!